pkg/auth/providers/azure/clients/ms_graph_client_test.go

```diff
@@ -0,0 +1,174 @@
+package clients
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+	mgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+)
+
+func TestAzureClient_connection_failures(t *testing.T) {
```

Comment:
This is testing the functionality inside the hamilton library. In my opinion it would have been enough to verify the hamilton library was called with the right parameters. This is not a blocking comment.

Reply:
resolving as it was clarified offline